internal/scan-manager/scanners/veracode/src/main/java/org/wso2/security/tools/scanmanager/scanners/veracode/service/VeracodeScanner.java

```diff
@@ -0,0 +1,304 @@
+
+        ResponseEntity responseEntity;
+        String result;
+
+        try {
+            result = uploadAPIWrapper.deleteBuild(scanContext.getAppId());
+            CallbackUtil.updateScanStatus(scanContext.getJobId(), ScanStatus.CANCELED, null,
+                    null);
+            if (VeracodeResultProcessor.isOperationProceedWithoutError(result)) {
+                responseEntity = new ResponseEntity<>(HttpStatus.ACCEPTED);
+            } else {
+                String message = "Error occured while deleting the last scan of the application : "
+                        + scanRequest.getAppId();
+                responseEntity = new ResponseEntity<>(new ErrorMessage(HttpStatus.BAD_REQUEST.value(),
+                        message), HttpStatus.BAD_REQUEST);
+                CallbackUtil.updateScanStatus(scanContext.getJobId(), ScanStatus.ERROR, null,
+                        null);
+                CallbackUtil.persistScanLog(scanContext.getJobId(), message, LogType.ERROR);
+            }
+        } catch (IOException e) {
+            String message = "Error occured while deleting the last scan of the application : "
+                    + scanRequest.getAppId() + VeracodeResultProcessor.getFullErrorMessage(e);
+            responseEntity = new ResponseEntity<>(new ErrorMessage(HttpStatus.INTERNAL_SERVER_ERROR.value(),
+                    message), HttpStatus.INTERNAL_SERVER_ERROR);
+            CallbackUtil.updateScanStatus(scanContext.getJobId(), ScanStatus.ERROR, null,
+                    null);
+            CallbackUtil.persistScanLog(scanContext.getJobId(), message, LogType.ERROR);
+        }
+        return responseEntity;
+    }
+
+    /**
+     * Build the upload wrap for Upload API.
+     *
+     * @param options represents the Veracode credentials and other required configurations
+     * @return Veracode Upload API Wrapper
+     * @throws UnsupportedEncodingException
+     */
+    private UploadAPIWrapper getUploadAPIWrapper(VeracodeCommand.Options options) throws UnsupportedEncodingException {
+        uploadAPIWrapper = new UploadAPIWrapper();
+        this.setUpWrapperCredentials(uploadAPIWrapper, options);
+
+        return uploadAPIWrapper;
+    }
+
+    /**
+     * Build the upload wrap for Result API.
+     *
+     * @param options represents the Veracode credentials and other required configurations
+     * @return Veracode Result API Wrapper
+     * @throws UnsupportedEncodingException
+     */
+    private ResultsAPIWrapper getResultAPIWrapper(VeracodeCommand.Options options) throws UnsupportedEncodingException {
+        resultsAPIWrapper = new ResultsAPIWrapper();
+        this.setUpWrapperCredentials(resultsAPIWrapper, options);
+
+        return resultsAPIWrapper;
+    }
+
+    /**
+     * Initiate the scan in the Veracode.
+     */
+    private void startVeracodeScan() {
+        ScanTask scanTask = new ScanTask(uploadAPIWrapper, scanContext, resultsAPIWrapper);
```

Comment:
We should not pass `uploadAPIWrapper` here. Instead the `ScanTask` should call the `VeracodeAPIUtil` and get an instance of it's own.